src/Microsoft.ML.StandardLearners/Standard/ModelStatistics.cs

```diff
@@ -15,24 +15,29 @@
 using Microsoft.ML.Model;
 
 // This is for deserialization from a model repository.
-[assembly: LoadableClass(typeof(LinearModelStatistics), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(LinearModelParameterStatistics), null, typeof(SignatureLoadModel),
     "Linear Model Statistics",
-    LinearModelStatistics.LoaderSignature)]
+    LinearModelParameterStatistics.LoaderSignature)]
+
+// This is for deserialization from a model repository.
+[assembly: LoadableClass(typeof(ModelStatistics), null, typeof(SignatureLoadModel),
+    "Model Statistics",
+   ModelStatistics.LoaderSignature)]
 
 namespace Microsoft.ML.Learners
 {
     /// <summary>
     /// Represents a coefficient statistics object.
     /// </summary>
-    public readonly struct CoefficientStatistics
+    public sealed class CoefficientStatistics
     {
         public readonly string Name;
```

Comment:
>Name [](start = 31, length = 4)

@tomfinley,  I haven't addressed your comment about removing the name from here, and operating on index. 
Since this is user facing, i was thinking that names might be easier to interpret? 

Do you think it would be enough to modify the API in line 373, that currently takes the FeatureColumn, to instead take the slot names of the metadata? The user would be in charge of extracting the slot names, and passing them to the API.

Reply:
But the names are not easier to interpret, because everywhere else we index a feature, we do so by name.

Reply:
I think I don't fully understand your point. Happy to make the change, just would like to understand the problem. 

The feature "names" are already saved in the annotations/metadata of the feature column. Since that is done, why should we not leave it to it, rather than convert to indexes where the users would have to know that bias is the first weight, etc. 

---
In reply to: [246904367](https://github.com/dotnet/machinelearning/pull/2048#discussion_r246904367) [](ancestors = 246904367)